editors/code/src/scopes.ts

```diff
@@ -0,0 +1,146 @@
+import * as fs from 'fs';
+import * as jsonc from 'jsonc-parser';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export interface TextMateRule {
+    scope: string | string[];
+    settings: TextMateRuleSettings;
+}
+
+export interface TextMateRuleSettings {
+    foreground: string | undefined;
+    background: string | undefined;
+    fontStyle: string | undefined;
+}
+
+// Current theme colors
+const rules = new Map<string, TextMateRuleSettings>();
+
+export function find(scope: string): TextMateRuleSettings | undefined {
+    return rules.get(scope);
+}
+
+// Load all textmate scopes in the currently active theme
+export function load() {
+    // Remove any previous theme
+    rules.clear();
+    // Find out current color theme
+    const themeName = vscode.workspace
+        .getConfiguration('workbench')
+        .get('colorTheme');
+
+    if (typeof themeName !== 'string') {
```

Comment:
Is this needed? I don't know what Code returns if the user doesn't respect the expected schema (e.g. `"workbench.colorTheme": {}`). But it doesn't hurt, I guess.

Reply:
I will look into the json schema and see what valid options they got there. If string is the only valid option then I can remove this.

Reply:
Ah, don't bother, it's not that important.